src/chat/chatAgentRequestHandler.ts

```diff
@@ -46,17 +47,24 @@ export const createSqlAgentRequestHandler = (
         return () => `conversationUri${idCounter++}`;
     })();
 
+    const getLogger = (() => {
+        const logger = Logger.create(vscodeWrapper.outputChannel, "MssqlCopilot");
+
+        return () => logger;
+    })();
+
     const handler: vscode.ChatRequestHandler = async (
         request: vscode.ChatRequest,
         chatContext: vscode.ChatContext,
         stream: vscode.ChatResponseStream,
         token: vscode.CancellationToken,
     ): Promise<ISqlChatResult> => {
         const correlationId = Utils.generateGuid();
+        const logger = getLogger();
         let conversationUri = getNextConversationUri();
         let connectionUri = vscodeWrapper.activeTextEditorUri;
-
-        Utils.logDebug(
+        logger.info("In handler");
```

Comment:
This "entered/exited XYZ function" can probably be verbose- instead of info-level logging.  Same for the other spots